lychee-bin/src/commands/check.rs

```diff
@@ -155,10 +156,33 @@ fn show_progress(
             pb.println(out);
         }
     } else {
-        if (response.status().is_success() || response.status().is_excluded()) && !verbose {
+        if !verbose && (response.status().is_success() || response.status().is_excluded()) {
```

Comment:
It's more idiomatic to write

if verbose || !(response.status.is_success() || response.status.is_excluded()) {
    // print
}

Reply:
To be honest I don't like any version so far. It's easy to get wrong.
For example, the correct expression of your version sould be 
```rust
if verbose || !(response.status().is_success() && response.status().is_excluded())
```
(There needs to be `&&` between success and excluded`.)


What do you think about splitting it up?

```rust
if !verbose {
    if response.status().is_success() || response.status().is_excluded() {
        return Ok(());
    }
}
        ```

Reply:
Super opinionated, but I honestly like the the two if conditions the easiest to reason about &mdash; it's just that clippy didn't like it. I experimented a bit but couldn't come up with anything that doesn't make my head spin, so I added a clippy exception for this single case.
We should probably refactor that part anyway at some point. 😉

Reply:
Will merge this now. We can revisit this at some point if we want to change it. At least the logic should be correct now and we have a unit test for the correct behavior. This should make future refactoring easier.

Reply:
my point is to eliminate a local return here.

So instead of (if condition return else print), it's much better to have (if condition print else ok)

Reply:
It could be even merged with the `} else {` into an `} else if ... {`:
```ru
    } else if verbose || !(response.status().is_success() || response.status().is_excluded()) {
        writeln!(output, "{out}")?;
    }
    Ok(())
```
The second `||` is correct here instead of `&&`. Probably better to understand:
```ru
    } else if verbose || (!response.status().is_success() && !response.status().is_excluded()) {
        writeln!(output, "{out}")?;
    }
    Ok(())
```
"Print if either verbose, or not successful and not excluded."

Reply:
Sure we can change it. Would you be interested in creating a pull request? It's not high priority, though.